src/state/sagas/datasets/FindAllDatasets/FindAllDatasets.js

```diff
@@ -8,11 +8,19 @@ import { STATUSES } from '../../../commons/Constants';
 
 export function* fetchAllDatasetsData(organizationId) {
   const { data } = yield call(Api.Datasets.findAllDatasets, organizationId);
+
   yield put({
     type: DATASET_ACTIONS_KEY.SET_ALL_DATASETS,
     list: data,
     status: STATUSES.SUCCESS,
   });
+
+  if (data?.length > 0) {
+    yield put({
+      type: DATASET_ACTIONS_KEY.SET_CURRENT_DATASET_INDEX,
+      selectedDatasetId: null,
+    });
+  }
```

Comment:
I'm not sure to understand :slightly_smiling_face: can we dispatch an action from another reducer action? if we directly set `selectedDatasetIndex` to `null` inside `SET_ALL_DATASETS` it won't dispatch `SET_CURRENT_DATASET_INDEX` and the first dataset won't be selected. We can optionally calculate its index inside `SET_ALL_DATASETS` but it will duplicate the same action we have in `SET_CURRENT_DATASET_INDEX` so it's not different from dispatching the action from the saga :thinking:

Reply:
Oh, my bad, I thought the goal of the lines above was simply to reset `selectedDatasetId` to `null` when the list is empty, but actually this dispatcher is used to select the first dataset in the list of _main_ datasets when the list is _not_ empty :upside_down_face: